File: Excel/451 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/451 Consecutive Numbers.xlsx", range = "A1:A20")
test  = read_excel("Excel/451 Consecutive Numbers.xlsx", range = "D1:E3")

result = input %>%
  mutate(group = cumsum(Numbers - lag(Numbers, default = 0) != 0),
         pos = ifelse(Numbers > 0, "P", "N")) %>%
  summarise(count = n() %>% as.numeric(), .by = c(group, Numbers, pos)) %>%
  filter(count == max(count), .by = pos) %>%
  summarise(Number = paste(unique(Numbers), collapse = ", "), Count = unique(count), .by = pos) %>%
  arrange(desc(Count)) %>%
  select(-pos)

identical(result, test)
# [1] TRUE

File: Excel/400-499/451/451 Challenge.py
import pandas as pd

input = pd.read_excel("451 Consecutive Numbers.xlsx", usecols="A")
test = pd.read_excel("451 Consecutive Numbers.xlsx",  usecols="D:E", nrows = 2)

result = input.assign(group=(input["Numbers"] != input["Numbers"].shift()).cumsum(),
                      pos=input["Numbers"].apply(lambda x: "P" if x > 0 else "N")) \
    .groupby(["group", "Numbers", "pos"]) \
    .size().reset_index(name="count") \
    .groupby("pos") \
    .apply(lambda x: x[x["count"] == x["count"].max()]) \
    .reset_index(drop=True) \
    .groupby("pos") \
    .agg(Number=("Numbers", lambda x: ", ".join(map(str, x.unique()))),
         Count=("count", "first")) \
    .sort_values("Count", ascending=False) \
    .reset_index(drop=True)

print(result)
print(test)